src/datastore/postgres-store.ts

```diff
@@ -5583,6 +5578,54 @@ export class PgDataStore
     });
   }
 
+  async getUnlockedAddressesAtBlock(block: DbBlock): Promise<StxUnlockEvent[]> {
+    return this.queryTx(async client => {
+      return await this.internalGetUnlockedAccountsAtHeight(client, block);
+    });
+  }
+
+  async internalGetUnlockedAccountsAtHeight(
+    client: ClientBase,
+    block: DbBlock
+  ): Promise<StxUnlockEvent[]> {
+    const current_burn_height = block.burn_block_height;
+    let previous_burn_height = current_burn_height;
+    if (block.block_height > 1) {
+      const previous_block = await this.getBlockByHeightInternal(client, block.block_height - 1);
+      if (previous_block.found) {
+        previous_burn_height = previous_block.result.burn_block_height;
+      }
+    }
+
+    const lockQuery = await client.query<{
+      locked_amount: string;
+      unlock_height: string;
+      block_height: string;
+      locked_address: string;
+      tx_id: Buffer;
+    }>(
+      `
+      SELECT locked_amount, unlock_height, block_height, tx_id, locked_address
+      FROM stx_lock_events
+      WHERE canonical = true AND unlock_height <= $1 AND unlock_height > $2
+      `,
+      [current_burn_height, previous_burn_height]
+    );
+
+    const result: StxUnlockEvent[] = [];
+    lockQuery.rows.forEach(row => {
+      const unlockEvent: StxUnlockEvent = {
```

Comment:
@asimm241 the code coverage report shows this path is never hit, along with pretty much all of the other Stacking `unlock` related code in `rosetta-helpers.ts` and `db-controller.ts`. Would you be able to write tests that cover these paths? I'm concerned about bugs around all these new functions.

Reply:
These have test code coverage now, great job!